File: src/main/java/com/senderman/lastkatkabot/command/user/LastPairs.java
package com.senderman.lastkatkabot.command.user;

import com.annimon.tgbotsmodule.api.methods.Methods;
import com.annimon.tgbotsmodule.services.CommonAbsSender;
import com.senderman.lastkatkabot.command.CommandExecutor;
import com.senderman.lastkatkabot.dbservice.ChatInfoService;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

@Component
public class LastPairs implements CommandExecutor {

    private final CommonAbsSender telegram;
    private final ChatInfoService chats;

    public LastPairs(CommonAbsSender telegram, ChatInfoService chats) {
        this.telegram = telegram;
        this.chats = chats;
    }

    @Override
    public String getTrigger() {
        return "/lastpairs";
    }

    @Override
    public String getDescription() {
        return "последние 10 пар чата";
    }

    @Override
    public void execute(Message message) {
        var chatId = message.getChatId();

        if (message.isUserMessage()) {
            Methods.sendMessage(chatId, "Команду нельзя использовать в ЛС!").callAsync(telegram);
            return;
        }

        var chatInfo = chats.findById(chatId);
        var pairs = chatInfo.getLastPairs();
        if (pairs == null || pairs.isEmpty()) {
            Methods.sendMessage(chatId, "В этом чате еще ни разу не запускали /pair!").callAsync(telegram);
            return;
        }

        var text = "<b>Последние 10 пар:</b>\n\n" + String.join("\n", pairs);
        Methods.sendMessage(chatId, text).callAsync(telegram);
    }
}

File: src/main/kotlin/com/senderman/lastkatkabot/usercommands/LastPairs.kt
package com.senderman.lastkatkabot.usercommands

import com.senderman.lastkatkabot.LastkatkaBotHandler
import com.senderman.lastkatkabot.Services
import com.senderman.neblib.CommandExecutor
import org.telegram.telegrambots.meta.api.objects.Message

class LastPairs(private val handler: LastkatkaBotHandler) : CommandExecutor {
    override val command: String
        get() = "/lastpairs"
    override val desc: String
        get() = "последние 10 пар чата"

    override fun execute(message: Message) {
        if (message.isUserMessage) return
        val chatId = message.chatId
        val history = Services.db.getPairsHistory(chatId)
        handler.sendMessage(chatId,
            history?.let {
                "<b>Последние 10 пар:</b>\n\n$it"
            } ?: "В этом чате еще никогда не запускали команду /pair!"
        )
    }
}